modules/gdscript/language_server/gdscript_text_document.cpp

```diff
@@ -112,12 +112,23 @@ void GDScriptTextDocument::didSave(const Variant &p_param) {
 		}
 
 		scr->update_exports();
-		ScriptEditor::get_singleton()->reload_scripts(true);
-		ScriptEditor::get_singleton()->update_docs_from_script(scr);
-		ScriptEditor::get_singleton()->trigger_live_script_reload(scr->get_path());
+
+		if ((bool)_EDITOR_GET("network/language_server/use_thread")) {
+			// It is not safe to call "reload_scripts" from the current thread, so it needs to be deferred
+			// otherwise it will randomly cause a crash.
+			(callable_mp(this, &GDScriptTextDocument::reload_script)).call_deferred(scr);
+		} else {
+			reload_script(scr);
+		}
```

Comment:
This should technically work and should cover all cases.

```suggestion
		if (!Thread::is_main_thread()) {
			callable_mp(this, &GDScriptTextDocument::reload_script).call_deferred(scr);
		} else {
			reload_script(scr);
		}
```